### PR TITLE
Update SEVIRI iCARE reader to support dask.

### PR DESCRIPTION
**Pull request by simonrp84**
The iCARE/SEVIRI reader was not daskified, data was being stored as numpy arrays. This raised an error when trying to create a `natural_color` image as the Rayleigh correction code was expecting dask arrays.

This PR updates the reader so that the data is stored as a dask array.

### CODE REVIEW

File: satpy/readers/seviri_l1b_icare.py [hunk 1]
```diff
@@ -259,7 +262,9 @@ def get_dataset(self, ds_id, ds_info):
             # Now we correct range from 0-1 to 0-100 for VIS:
             if ds_id['name'] in self.ref_bands:
                 data.values *= 100.
-        return data
+        variable = xr.DataArray(da.from_array(data, chunks=(CHUNK_SIZE, CHUNK_SIZE)),
+                                dims=['y', 'x'])
```
Comment (mraspaud) [inline satpy/readers/seviri_l1b_icare.py:266]: I think the problem comes from a few lines above with: `data.values = data.values.astype(np.float32)`. This looks like it converts the daskified DataArray to numpy. Doesn't `data = data.astype(np.float32)` work instead?

**Review by mraspaud — CHANGES_REQUESTED** [review]
Thanks for the quick fix. I think this could be addressed a bit differently though...

File: satpy/readers/seviri_l1b_icare.py (hunk 1 above)
Comment (simonrp84) [reply satpy/readers/seviri_l1b_icare.py:266]: No, that doesn't work as the data returned from the HDF4 file is a numpy array, it has to be explicitly converted to dask / xarray.
That said, I didn't need `.values` in the code so have at least deleted that :)

**Comment by simonrp84** [comment]
Rebasing my branch makes git sad :( Will do this thing again.

**Pull request closed without merging**